worlds/ladx/ForeignItemIcons.py

```diff
@@ -0,0 +1,896 @@
+import re
+from .LADXR.locations.items import *
+from .LADXR.locations.constants import CHEST_ITEMS
+
+
+class ForeignItemIconMatcher:
+    name_cache = {}
+
+    def __init__(self):
+        # Build the generic name cache, using LADX item names and SYNONYMS.
+        # The item names are broken into parts, and pluralized versions are also inserted.
+        for item in CHEST_ITEMS.keys():
+            self.name_cache[item] = item
+            for word in item.split("_"):
+                if word not in BLOCKED_ASSOCIATIONS and not word[-1].isnumeric():
+                    self.name_cache[word] = item
+        for name in SYNONYMS.values():
+            assert name in self.name_cache, name
+            assert name in CHEST_ITEMS, name
+        self.name_cache.update(SYNONYMS)
+        pluralizations = {k + "S": v for k, v in self.name_cache.items()}
+        self.name_cache = pluralizations | self.name_cache
+
+
+    def get_icon_for_other_world(self, foreign_item_name:str, foreign_game:str) -> str:
```

Comment:
```suggestion
    def get_icon_for_other_world(self, foreign_item_name: str, foreign_game: str) -> str:
```
Minor clean up, things like this you are free to ignore